Client/Source/Client.cpp

```diff
@@ -0,0 +1,361 @@
+                }
+                else if (_handler.getCurrentState() == Handler::State::PARTIAL)
+                {
+                    sendCoordinates(_trajectoryManager.partialMovement(_handler.getParsedResult()));
+                }
+                else if (_handler.getCurrentState() == Handler::State::HOME)
+                {
+                    sendCoordinates(RobotData::getDefaultPosition());
+                }
+                else if (_handler.getCurrentState() == Handler::State::FROM_FILE)
+                {
+                    _robotData = _logger.readLine<RobotData>();
+                    if (!_logger.hasAnyInputErrors())
+                    {
+                        sendCoordinates(_robotData);
+                    }
+                    else
+                    {
+                        _printer.writeLine(std::cout, "ERROR 05: Some error occurred in input "
+                                           "stream! Input stream will be restarted.");
+                        _logger.restartStream(logger::Logger::TypeStream::INPUT_STREAM);
+                    }
+                }
+                else if (_handler.getCurrentState() != Handler::State::DEFAULT)
+                {
+                    sendCoordinates(_robotData);
+                }
+                else if (input == "test")
+                {
+                    _robotData = RobotData::getDefaultPosition();
+                    constexpr int kNumberOfIterations = 1000;
+                    constexpr int kDefaultMultiplier  = 100;
+                    for (int i = 1; i <= kNumberOfIterations; ++i)
+                    {
+                        _robotData.coordinates.at(Handler::Y) += kDefaultMultiplier * i
+                                                              * (i & 1 ? 1 : -1);
+                        sendCoordinates(_robotData);
+                    }
+                }
+                break;
+
+            default:
+                assert(false);
+                break;
+        }
+    }
+}
+
+std::chrono::duration<double> Client::getDuration() const noexcept
+{
+    return _duration;
+}
+
+RobotData Client::getRobotData() const noexcept
+{
+    return _robotData;
+}
+
+bool Client::tryConnect(const int port, const std::string& ip, QTcpSocket* socketToConnect,
+                        const bool isNeedToSendSystem, const int msecs) const
+{
+    if (socketToConnect->isOpen())
+    {
+        socketToConnect->close();
+    }
+
+    socketToConnect->connectToHost(ip.c_str(), port);
+
+    if (socketToConnect->waitForConnected(msecs))
+    {
+        _printer.writeLine(std::cout, "Connected to Server!");
+        if (isNeedToSendSystem)
+        {
+            sendCoordinateSystem(CoordinateSystem::WORLD);
+        }
+        
+        return true;
+    }
+
+    _printer.writeLine(std::cout, "Not connected to Server!");
+    return false;
+}
+
+void Client::tryReconnect()
+{
+    launch();
+}
```

Comment:
Do you need this method?

Reply:
Yes, because not every reconnection is equal to connection. And I tried to highlight the main methods for the future network interface.